gst/nnstreamer/tensor_query/tensor_query_common.c

```diff
@@ -66,26 +73,447 @@ typedef struct
   };
 } TensorQueryClientData;
 
+/* TCP methods */
+static gboolean query_server_socket_establish (TensorQueryServerData *
+    server_data, const gchar * host, guint16 port);
+static gboolean query_server_tcp_open (TensorQueryServerData * server_data);
+static gboolean query_server_tcp_receive (TensorQueryServerData * server_data,
+    gchar * data, gsize size);
+static gboolean query_server_tcp_send (TensorQueryServerData * server_data,
+    gchar * data, gsize size);
+static gboolean query_server_tcp_close (TensorQueryServerData * server_data);
+
 /**
- * @brief Connect to the specified address.
+ * @brief return initialized server_data
+ * @param src_info tensors info shared with serversrc element
  */
-int
-nnstreamer_query_connect (uint64_t id, const char *ip, uint32_t port,
-    uint32_t timeout_ms)
+void *
+gst_query_server_data_init (GstTensorsInfo * src_info)
 {
-  /** @todo remove "UNUSED" when you implement the full features */
-  UNUSED (id);
-  UNUSED (ip);
-  UNUSED (port);
-  UNUSED (timeout_ms);
-
-  /** NYI: To avoid `defined but not used` */
-  if (NULL == table) {
+  TensorQueryServerData *server_data = g_new (TensorQueryServerData, 1);
+  server_data->protocol = _TENSOR_QUERY_PROTOCOL_END;
+  server_data->src_info = src_info;
+
+  server_data->current_client_id = NULL;
```

Comment:
It should be checked and freed in `gst_query_server_data_free`

Reply:
Fixed!! Thank you